### ISSUE TITLE
$transaction weird behaviour

### ISSUE THREAD

**Issue by jwld** [prisma/prisma#21631]
### Bug description

```
const [emailConflict, phoneConflict] = await prisma.$transaction([
  prisma.worker.findUnique({ where: { email: data.email } }),
  prisma.worker.findUnique({ where: { phone: data.phone } })
])
```

Both are `null`, even when there's an email conflict. If I take the second query out:

```
const [emailConflict] = await prisma.$transaction([
  prisma.worker.findUnique({ where: { email: data.email } })
])
```

then `emailConflict` is not null and returns the conflicting worker as expected.

### How to reproduce

As above

### Prisma information

```
model Worker {
  email     String  @unique
  phone     String  @unique
  id        String   @id @default(cuid())
}
```

### Environment & setup

- OS: macOS
- Database: PostgreSQL
- Node.js version: 18.9.0

### Prisma Version

```
5.5.2
```

**Comment by Murtatrxx** [comment]
Do you use driver adapters, because they also cause weird problems in my case (#21592)

**Comment by janpio** [comment]
What do you mean by "even when there's an email conflict" @jwld?

**Comment by jwld** [comment]
@Murtatrxx nope, nothing like that

@janpio Sorry, I mean if there's a worker with that email in the DB already. For context we run this to check for potential conflicts before creating a new worker, rather than letting prisma throw a unique conflict error.

**Comment by jkomyno** [comment]
Hi @jwld, I confirm that this issue was happening in Prisma version you reported, 5.5.2.

We actually fixed this problem in [Prisma 5.12.0](https://github.com/prisma/prisma/releases/tag/5.12.0), albeit not explicitly (it was likely a by-product of the fix for https://github.com/prisma/prisma/issues/23343).
I hence added explicit tests for the scenario you highlighted in [here](https://github.com/prisma/prisma/pull/24764), to ensure no future user will stumble upon the same problem.

Thanks!

**Comment by jwld** [comment]
Thanks for the update, good to hear!
